src/scenarios/helloios/pre.py

```diff
@@ -1,10 +1,53 @@
 '''
 pre-command
 '''
-from performance.logger import setup_loggers
-from shutil import copytree
+import sys
+import os
+from zipfile import ZipFile
+from performance.logger import setup_loggers, getLogger
+from shutil import copyfile, copytree, move
 from shared.const import PUBDIR
+from argparse import ArgumentParser
 
 setup_loggers(True)
 
-copytree('app', PUBDIR)
+parser = ArgumentParser()
+parser.add_argument('--unzip', help='Unzip ipa file and report extracted tree', action='store_true', default=False)
+parser.add_argument(
+        '--name',
+        dest='name',
+        required=True,
+        type=str,
+        help='Name of the file/folder to setup (with .app or .ipa)')
+args = parser.parse_args()
+
+name = args.name
+namezip = '%s.zip' % (name)
+if not os.path.exists(PUBDIR):
+    os.mkdir(PUBDIR)
+if not os.path.exists(name):
+    getLogger().error('Cannot find %s' % (name))
+    exit(-1)
+if args.unzip:
+    if not os.path.exists(namezip):
+        copyfile(name, namezip)
+
+    with ZipFile(namezip) as zip:
+        zip.extractall(os.path.join('.', PUBDIR))
+
+    baseDir = os.path.join(PUBDIR, 'Payload')
+    appFolder = os.listdir(baseDir) # Should only be one
+    if len(appFolder) > 1:
+        getLogger.error("More than one app folder found in payload! Should only be one.")
+        exit(-1)
+
+    appFolderPath = os.path.join(baseDir, appFolder[0])
+    for subfile in os.listdir(appFolderPath):
```

Comment:
Why are we moving the extracted files after unzipping rather than just measuring the size where we unpack them?

Reply:
The main reason was because the structure of the unzipped ipa is folder.app/{interesting files}. I can play around with it a little bit more to and see if I can get the dirs argument to set the directory to parse to work instead though.

Reply:
I double checked some assumptions I made about the ipa file structure and they were incorrect so I removed the moving section to match measuring the whole zip file.